File: server/model/Todo.model.js
const mongoose = require('mongoose');
const bcrypt = require('bcrypt');
const db = require('../config/db');
const UserModel = require('../model/User.model');

const {Schema, model} = mongoose;

const todoSchema = new  Schema({
    userId:{
        type:Schema.Types.ObjectId,
        ref: UserModel.modelName
    },
    title:{
       type:String,
       required:true,

    },
    description: {
        type : String,
        required:true,
    },
});

const todoModel = model('Todo',todoSchema);

module.exports = todoModel;

File: lib/models/todo_model.dart
class Todo {
  String title;
  String description;
  String id;

  Todo({
    required this.title,
    required this.description,
    required this.id,
  });

  factory Todo.fromJson(Map<String, dynamic> json) {
    return Todo(
      title: json['title'],
      description: json['description'],
      id: json['_id'],
    );
  }
}
